client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
```

Comment:
javascript decorators are still stage-1, but if we had them available, I could see us keeping this logic closer together and annotating the class method so that it registers with this middleware

Maybe something very roughly like:
```javascript
@createNotice( POST_SAVE_SUCCESS )
postDeleteSuccess( action, dispatch ) {
    dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
}
```